api-processor/src/main/java/com/synopsys/integration/alert/processor/api/NotificationContentProcessor.java

```diff
@@ -43,28 +43,34 @@ public NotificationContentProcessor(
-                                                                       .reduce(ProcessedProviderMessageHolder::reduce)
-                                                                       .orElse(ProcessedProviderMessageHolder.empty());
+            .stream()
+            .map(providerMessageExtractionDelegator::extract)
+            .reduce(ProcessedProviderMessageHolder::reduce)
+            .orElse(ProcessedProviderMessageHolder.empty());
         return processExtractedNotifications(processingType, extractedProviderMessages);
     }
 
     private ProcessedProviderMessageHolder processExtractedNotifications(ProcessingType processingType, ProcessedProviderMessageHolder providerMessages) {
+        List<ProcessedProviderMessage<ProjectMessage>> filteredProjectMessages = filterProcessedMessages(providerMessages.getProcessedProjectMessages());
+        List<ProcessedProviderMessage<SimpleMessage>> filteredSimpleMessages = filterProcessedMessages(providerMessages.getProcessedSimpleMessages());
         if (ProcessingType.DEFAULT.equals(processingType)) {
-            return providerMessages;
+            return new ProcessedProviderMessageHolder(filteredProjectMessages, filteredSimpleMessages);
         }
 
-        List<ProcessedProviderMessage<ProjectMessage>> digestedMessages = projectMessageDigester.digest(providerMessages.getProcessedProjectMessages());
+        List<ProcessedProviderMessage<ProjectMessage>> digestedMessages = projectMessageDigester.digest(filteredProjectMessages);
         if (ProcessingType.SUMMARY.equals(processingType)) {
             List<ProcessedProviderMessage<SimpleMessage>> summarizedMessages = digestedMessages
-                                                                                   .stream()
-                                                                                   .map(projectMessageSummarizer::summarize)
-                                                                                   .collect(Collectors.toList());
-            List<ProcessedProviderMessage<SimpleMessage>> allSimpleMessages = ListUtils.union(providerMessages.getProcessedSimpleMessages(), summarizedMessages);
+                .stream()
+                .map(projectMessageSummarizer::summarize)
+                .collect(Collectors.toList());
+            List<ProcessedProviderMessage<SimpleMessage>> allSimpleMessages = ListUtils.union(filteredSimpleMessages, summarizedMessages);
             return new ProcessedProviderMessageHolder(List.of(), allSimpleMessages);
         }
-        return new ProcessedProviderMessageHolder(digestedMessages, providerMessages.getProcessedSimpleMessages());
+        return new ProcessedProviderMessageHolder(digestedMessages, filteredSimpleMessages);
+    }
+
+    private <T> List<T> filterProcessedMessages(List<T> processedProviderMessages) {
+        return processedProviderMessages.stream().distinct().collect(Collectors.toList());
```

Comment:
Is there a reason we don't just change ProcessedProviderMessageHolder to use Set's instead of Lists?

Reply:
I believe we have duplicate ProjectMessage objects because of the vulnerabilities coming from different origins. This filtering probably should go in the reduce method of the ProcessedProviderMessageHolder since it removes duplicate origin data.

Reply:
I left this as a list as all the classes and models that reference this data use list. My original thought was set but didn't want to go around changing everything incase that broke something